### PR TITLE
Replace python and node calls with native bash

### PR DESCRIPTION
**Pull request by syedhassaanahmed**
Purpose of this PR is to
- Use `jq` for json parsing instead of python calls
- Generate Event Hub SAS token in bash instead of `azure-iot-common` Node lib

*Note:* Instead of keeping SAS token generation code directly in `run.sh`, I've experimented with `source` command and loaded the function `get_sas_token()` from [my gist](https://gist.github.com/syedhassaanahmed/bf05bbff8d9696fb978c586881dda61f).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,8 +6,8 @@ before_install:
 - sudo apt-key adv --keyserver packages.microsoft.com --recv-keys 417A0893
 - sudo apt-get install -y apt-transport-https
 - sudo apt-get update && sudo apt-get install -y azure-cli
+- sudo apt-get install -y jq
 install:
-- npm install
```
Comment (vjrantal) [inline .travis.yml:10]: Why was this removed?

File: README.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ For more details and some background information about this project, see the blo
 
 ## Prerequisites
 
-* Run `npm install` to install dependencies required by the node script
+* Run `sudo apt-get install -y jq` to install [jq](https://stedolan.github.io/jq/) required by the bash script
```
Comment (vjrantal) [inline README.md]: Could you add a note that if on MacOS, jq can be installed with `brew install jq`?

Also, a new dependency is that openssl is installed and found from the path. This could be mentioned.

File: run.sh [hunk 3]
```diff
@@ -40,7 +35,15 @@ SLEEP_IN_SECONDS=${SLEEP_IN_SECONDS:=300}
...
 WRK_OPTIONS=${WRK_OPTIONS:="-d $((SLEEP_IN_SECONDS*2))"}
-WRK_HEADER=$(node -e "require('./index.js').printHeader()")
+
+EVENT_HUB_SHARED_ACCESS_KEY_NAME=$(echo $(echo $EVENT_HUB_CONNECTION | cut -d";" -f2) | cut -d"=" -f2)
+
+EVENT_HUB_SHARED_ACCESS_KEY=$(echo $EVENT_HUB_CONNECTION | cut -d";" -f3)
+EVENT_HUB_SHARED_ACCESS_KEY=${EVENT_HUB_SHARED_ACCESS_KEY#SharedAccessKey=}
+
+# generate SAS token using remote helper code, by default token expires 24 hours from now
+wget -O /tmp/get-sas-token.sh -q https://gist.githubusercontent.com/syedhassaanahmed/bf05bbff8d9696fb978c586881dda61f/raw/42e5bbf7ef75cb3f44b82b6b95dc77303f7e23d0/get-sas-token.sh && source /tmp/get-sas-token.sh && rm -f /tmp/get-sas-token.sh
```
Comment (vjrantal) [inline run.sh]: I would propose that you'd add the script `get-sas-token.sh` into this repository and call it locally instead of wget'ting it. What do you think?

The script should have:

```
set -e
set -o pipefail
```

so that if the sub-commands fail, the script itself fails as well.

**Review by vjrantal — COMMENTED** [review]
The external script did not work on MacOS and failed to error:

```
openssl:Error: 'sha256' is an invalid command
```

This might be related https://stackoverflow.com/a/8510087 . I can check soon what is the right command to use on MacOS.

File: .travis.yml (hunk 1 above)
Comment (vjrantal) [reply .travis.yml:10]: Ah, of course, because we don't have any npm dependencies anymore :)

File: README.md (hunk 2 above)
Comment (syedhassaanahmed) [reply README.md]: Done, I've added link to jq's download page so that appropriate instructions can be followed for each OS.

File: .travis.yml (hunk 1 above)
Comment (syedhassaanahmed) [reply .travis.yml:10]: Correct :)

File: run.sh (hunk 3 above)
Comment (syedhassaanahmed) [reply run.sh]: Done

**Comment by syedhassaanahmed** [comment]
Now the SAS token is generated in Python instead of using OpenSSL from bash.

**Comment by vjrantal** [comment]
Tested locally on MacOS and here https://travis-ci.org/vjrantal/event-hub-loader/builds/330234779?utm_source=github_status&utm_medium=notification on Ubuntu and works as expected.

Thanks for the contribution! 👍
